hls4ml/templates/quartus/firmware/nnet_utils/nnet_padding_stream.h

```diff
@@ -0,0 +1,87 @@
+#ifndef NNET_PADDING_STREAM_H_
+#define NNET_PADDING_STREAM_H_
+
+namespace nnet {
+
+template<class res_T, typename CONFIG_T>
+inline void fill_zero(stream<res_T> &res) {
+    hls_register res_T res_part;
+	#pragma unroll
+    for (int i = 0; i < CONFIG_T::n_chan; i++) {
+	    res_part[i] = 0;
+    }
+    res.write(res_part);
+}
+
+template<class data_T, class res_T, typename CONFIG_T>
+inline void fill_data(stream<data_T> &data, stream<res_T> &res) {
+    hls_register data_T data_part = data.read();
+    hls_register res_T res_part;
+    #pragma unroll
+    for (int i = 0; i < CONFIG_T::n_chan; i++) {
+        res_part[i] = data_part[i];
+    }
+    res.write(res_part);
+}
+
+template<class data_T, class res_T, typename CONFIG_T>
+void zeropad1d_cl(stream<data_T> &data, stream<res_T> &res) {
+    PadLeft: 
+    for (int i = 0; i < CONFIG_T::pad_left; i++) {
+        fill_zero<res_T, CONFIG_T>(res);
+    }
+
+    CopyMain: 
+    for (int i = 0; i < CONFIG_T::in_width; i++) {
+        fill_data<data_T, res_T, CONFIG_T>(data, res);
+    }
+
+    PadRight: 
+    for (int i = 0; i < CONFIG_T::pad_right; i++) {
+        fill_zero<res_T, CONFIG_T>(res);
+    }
+}
+
+template<class data_T, class res_T, typename CONFIG_T>
+void zeropad2d_cl(stream<data_T> &data, stream<res_T> &res) {
+    #pragma loop_coalesce 2
```

Comment:
Synthesis complained about the loop_coalesce pragma coming before the label. I think it needs to be label, pragma, loop, not pragma, label, loop.